src/Pact/Native.hs

```diff
@@ -1632,15 +1641,41 @@ keccak256Def = defGasRNative
 
 hyperlaneDefs :: NativeModule
 hyperlaneDefs = ("Hyperlane",)
-  [ hyperlaneMessageIdDef
-  , hyperlaneDecodeTokenMessageDef
+  [ hyperlaneMessageIdDef TyAny
+  , hyperlaneDecodeTokenMessageDef TyAny
+  ]
+
+hyperlaneAmendedDefs :: NativeModule
+hyperlaneAmendedDefs = ("Hyperlane",)
+  [ hyperlaneMessageIdDef (TyUser (snd hyperlaneDataSchema))
+  , hyperlaneDecodeTokenMessageDef (TyUser (snd hyperlaneDataSchema))
+  , tokenMessageERC20Schema
+  , hyperlaneDataSchema
   ]
 
-hyperlaneMessageIdDef :: NativeDef
-hyperlaneMessageIdDef = defGasRNative
+tokenMessageERC20Schema :: NativeDef
+tokenMessageERC20Schema = defSchema "hyperlane-token-erc20"
+  "Schema type for ERC20 TokenMessage"
+  [(FieldKey "recipient", tTyString)
+  ,(FieldKey "amount", tTyInteger)]
+
+hyperlaneDataSchema :: NativeDef
+hyperlaneDataSchema = defSchema "hyperlane-token-msg"
+  "Schema type for hyperlane messages"
+    [ (FieldKey "version", tTyInteger)
+    , (FieldKey "nonce", tTyInteger)
+    , (FieldKey "originDomain", tTyInteger)
+    , (FieldKey "destinationDomain", tTyInteger)
+    , (FieldKey "recipient", tTyString)
+    , (FieldKey "tokenMessage", tTyObject (TyUser (snd tokenMessageERC20Schema)))
```

Comment:
Just to confirm @chessai or @edmundnoble should this or should this not contain a `chainId` field?